src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -926,6 +934,9 @@ void VideoDecoder::convertAVFrameToDecodedOutputOnCPU(
         streamInfo.colorConversionLibrary ==
         ColorConversionLibrary::FILTERGRAPH) {
       output.frame = convertFrameToTensorUsingFilterGraph(streamIndex, frame);
+      if (preAllocatedOutputTensor.has_value()) {
+        preAllocatedOutputTensor.value().copy_(output.frame);
```

Comment:
@ahmadsharif1 you suggested

> Can you add a TODO to write a test that confirms the data_ptr in the pre-allocated tensor is the same as the return value in both color-conversion library cases? Otherwise this can regress and we would not know any better.


Just to clarify, you're suggesting to assert that `output.data_ptr() == preAllocatedOutputTensor.data_ptr()` ?

I think this will hold for swscale, but not for filtegraph. In the `copy_` above my understanding is that we copy the *data* at `output.data_ptr() ` into `preAllocatedOutputTensor.data_ptr()`, but the pointers themselves may live at different locations.

I guess that means that the caller should *not* rely on the `output.frame` if a pre-allocated tensor was passed. Are we OK with that?